ims/einstein/ceph.py

```diff
@@ -186,6 +242,21 @@ def snap_unprotect(self, img_id, snap_name):
             raise file_system_exceptions.ImageBusyException(img_id)
 
     @log
+    def is_snap_protected(self, img_id, snap_name):
+        """
+        Find out whether a snapshot is protected from deletion
+
+        :param img_id: what the image is called
+        :param snap_name: the snapshot to check
+        :return: bool- whether the snapshot is protected
+        """
+        try:
+            with self.__open_image(img_id) as img:
+                return img.is_protected_snap(snap_name)
```

Comment:
@ravisantoshgudimetla this doesn't return if snap_name doesn't exist, instead we raise the ImageNotFound Exception

https://github.com/xukan/ims/blob/3ef299b2ea031556003edf9c29d5a86b0c49c5cd/ims/einstein/ceph.py#L232-L234